cli/lib/tasks/verify.js

```diff
@@ -191,37 +190,69 @@ const start = (options = {}) => {
 
   const packageVersion = util.pkgVersion()
   let binaryDir = state.getBinaryDir(packageVersion)
-  if (process.env.CYPRESS_BINARY_FOLDER) {
-    const envBinaryDir = process.env.CYPRESS_BINARY_FOLDER
-    logger.warn(stripIndent`
-    ${logSymbols.warning} Warning: You have set the environment variable: ${chalk.white('CYPRESS_BINARY_FOLDER=')}${chalk.cyan(envBinaryDir)}
-    
-      This overrides the default Cypress binary version used.
-    `)
-    logger.log()
-    binaryDir = envBinaryDir
-  }
 
   _.defaults(options, {
     force: false,
     welcomeMessage: true,
   })
 
-  return isMissingExecutable(binaryDir)
+  const checkEnvVar = () => {
+    debug('checking env variables')
+    if (process.env.CYPRESS_RUN_BINARY) {
+      const envBinaryPath = process.env.CYPRESS_RUN_BINARY
+      debug('CYPRESS_RUN_BINARY exists, =', envBinaryPath)
+      logger.warn(stripIndent`
+        ${logSymbols.warning} Warning: You have set the environment variable: ${chalk.white('CYPRESS_RUN_BINARY=')}${chalk.cyan(envBinaryPath)}
+        
+          This overrides the default Cypress binary version used.
+        `)
+      logger.log()
+
+      return util.isExecutableAsync(envBinaryPath)
+      .then((isExecutable) => {
+        debug('CYPRESS_RUN_BINARY is executable? :', isExecutable)
+        if (!isExecutable) {
+          return throwFormErrorText(errors.CYPRESS_RUN_BINARY.notValid(envBinaryPath))(stripIndent`
+          The supplied binary path is not executable
```

Comment:
Is there any guidance we can give them here on how to move forward, like "Provide the binary in x format" or something?

Reply:
Here's the full error printed to the user: 
![image](https://user-images.githubusercontent.com/14625260/40241228-783d3908-5a88-11e8-96db-5f4c51664830.png)

Reply:
similarly, win32:
![image](https://user-images.githubusercontent.com/14625260/40241696-a40ed20c-5a89-11e8-9426-a87dcf2c2472.png)